packages/safe-tools-client/app/services/hub-authentication.ts

```diff
@@ -48,27 +49,38 @@ export default class HubAuthenticationService extends Service {
     return await getSDK('HubAuth', ethersProvider, config.hubUrl);
   }
 
+  get authTokenKey() {
+    return `authToken-${this.wallet.address}-${this.network.chainId}`;
+  }
+
   get authToken(): string | null {
-    return this.storage.getItem('authToken');
+    return this.storage.getItem(this.authTokenKey);
   }
 
   set authToken(val: string | null) {
     if (val) {
-      this.storage.setItem('authToken', val);
+      this.storage.setItem(this.authTokenKey, val);
     } else {
       this.isAuthenticated = false;
-      this.storage.removeItem('authToken');
+      this.storage.removeItem(this.authTokenKey);
     }
   }
 
   async hasValidAuthentication() {
     const hubAuth = await this.getHubAuth();
-
-    return Boolean(
-      this.wallet.isConnected &&
-        this.authToken &&
-        (await hubAuth.checkValidAuth(this.authToken))
-    );
+    try {
+      const authAddress = this.authToken
+        ? await hubAuth.getAddress(this.authToken)
```

Comment:
I'm confused why fetching the address from the hub is needed. Why not read the address from the wallet service, construct the auth storage key, use it to get the auth token from the storage and perform `checkValidAuth(authToken)`?

Reply:
How do we validate the auth token belongs to which address?

Reply:
Doesn't `await hubAuth.checkValidAuth(this.authToken)` suffice to see if the token is valid?

Reply:
Only checking whether the auth token is valid or not. Not checking is that token is used by the right address or not.

Reply:
I see, you're addressing a case where someone could steal your token and plant it in their own localstorage, and prevent communicating to the hub in case the wallet address doesn't match the address encrypted in the token. 

I guess it doesn't hurt to have that extra layer of security.  However that will only work for until the token expires (1 day). And the attacker can easily use the leaked token in curl or use some other http client until the token expires

Reply:
Removed the getAddress function.